packages/lit-dev-content/src/components/litdev-search.ts

```diff
@@ -322,69 +361,28 @@ export class LitDevSearch extends LitElement {
     `;
   }
 
-  /**
-   * Load and deserialize search index into `LitDevSearch.siteSearchIndex`.
-   */
-  private _loadSearchIndex(): Promise<Minisearch<UserFacingPageData>> {
-    // We already have a search index, or a load request is in progress due to
-    // another component on the page requesting it.
-    if (
-      LitDevSearch._siteSearchIndex !== null ||
-      LitDevSearch._loadingSearchIndex
-    ) {
-      return LitDevSearch._loadingSearchIndex!;
-    }
-
-    // This must be done as chained promise because we need to synchronously
-    // set the static LitDevSearch._loadingSearchIndex promise.
-    const searchIndexPromise = fetch('/searchIndex.json')
-      .then((res) => res.text())
-      .then((searchIndexJson) => {
-        // Minisearch intialization config must exactly match
-        // `/lit-dev-tools-cjs/src/search/plugin.ts` Minisearch options.
-        LitDevSearch._siteSearchIndex = Minisearch.loadJSON<UserFacingPageData>(
-          searchIndexJson,
-          {
-            idField: 'id',
-            fields: ['title', 'heading', 'text'],
-            storeFields: ['title', 'heading', 'relativeUrl', 'isSubsection'],
-            searchOptions: {
-              boost: {title: 1.4, heading: 1.2, text: 1},
-              prefix: true,
-              fuzzy: 0.2,
-            },
-          }
-        );
-
-        return LitDevSearch._siteSearchIndex;
-      });
-    LitDevSearch._loadingSearchIndex = searchIndexPromise;
-
-    return searchIndexPromise;
-  }
-
   /**
    * Load the search index.
    */
   async firstUpdated() {
     // required for popping up on hydration
     this._isFocused = !!this.shadowRoot?.activeElement;
-    await this._loadSearchIndex();
-    // If a search query has already been written, fill suggestions.
-    this._querySearch(this._searchText);
+    // Initialize the search task only on the client and not the server.
+    this._searchTask = new Task(
```

Comment:
Very, very much optional, but as an exercise in composing controllers and Task, it might be nice to see an AlgoliaSearchController that composes task and builds-in the search API function and _algoliaClient

Reply:
I did it. LMK what you think. It could not be a true controller because it does not need any of the interface